enumerable_methods.rb

```diff
@@ -0,0 +1,100 @@
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
+        yield(self[i])
+        i += 1
+      end
+    else
+      'No block given'
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < length
+        yield(self[i], i)
+        i += 1
+      end
+    else
+      'No block given'
+    end
+  end
+
+  def my_select
+    ary = []
+    if block_given?
+      my_each { |elem| ary.push(elem) if yield(elem) }
+    else
+      'No block given'
+    end
+    ary
+  end
+
+  def my_all?
+    if block_given?
+      my_each { |elem| return false unless yield(elem) }
+    else
+      my_each { |elem| return false unless elem }
+    end
+    true
+  end
+
+  def my_any?
+    if block_given?
+      my_each { |elem| return true if yield(elem) }
+    else
+      my_each { |elem| return true if elem }
+    end
+    false
+  end
+
+  def my_none?
+    if block_given?
+      my_each { |elem| return false if yield(elem) }
+    else
+      my_each { |elem| return false if elem }
+    end
+    true
+  end
+
+  def my_count
+    c = 0
+    if block_given?
+      my_each { |elem| c += 1 if yield(elem) }
+    else
+      my_each { c += 1 }
+    end
+    c
+  end
+
+  def my_map(proc = nil)
+    if proc
+      ary = []
+      my_each { |elem| ary.push(proc.call(elem)) }
+      ary
+    elsif block_given?
+      ary = []
+      my_each { |elem| ary.push(yield(elem)) }
+      ary
+    else
+      to_enum(:my_map)
+    end
+  end
+
+  def my_inject(cont = nil)
+    if block_given?
+      cont ||= 0
+      my_each { |elem| cont = yield(cont, elem) }
+      cont
+    else
+      'No block Given (LocalJumpError)'
+    end
+  end
+end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method